ui/app/models/ssh/ca-config.js

```diff
@@ -5,16 +5,49 @@
 
 import Model, { attr } from '@ember-data/model';
 import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import { withModelValidations } from 'vault/decorators/model-validations';
 
+const validations = {
+  publicKey: [
+    {
+      validator(model) {
+        const { publicKey, privateKey, generateSigningKey } = model;
+        // if generateSigningKey is false, use the second validation to check if both public and private keys are set
+        return (publicKey && privateKey) || (!publicKey && !privateKey) || !generateSigningKey ? true : false;
+      },
+      message: 'Public Key and Private Key are both required if one of them is set.',
```

Comment:
Nit: I noticed in some places key is capitalized and others it's not? It seems to me that it should be `Public key` because it's not a proper noun but I'll defer to you

Reply:
After updating, there remains one uppercase use of it in the ssh-sign model. out of scope, but noting here.